network/p2p/sender.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var (
+	_ AppSender = (*Sender)(nil)
+	_ AppSender = (*FakeSender)(nil)
+)
+
+type AppRequestSender interface {
+	// SendAppRequest sends an AppRequest message to a peer
+	SendAppRequest(ctx context.Context, nodeID ids.NodeID, requestID uint32, bytes []byte) error
+}
+
+type AppResponseSender interface {
+	// SendAppRequest sends an AppResponse message to a peer
+	SendAppResponse(ctx context.Context, nodeID ids.NodeID, requestID uint32, bytes []byte) error
+}
+
+type AppGossipSender interface {
+	// SendAppGossip sends an AppGossip message to an arbitrary peer
+	SendAppGossip(ctx context.Context, bytes []byte) error
+	// SendAppGossipSpecific sends an AppGossip message to a given peer
+	SendAppGossipSpecific(ctx context.Context, nodeID ids.NodeID, bytes []byte) error
```

Comment:
Why not pass in a set of node IDs, since we do `set.Of` later?

Reply:
+1